server/fileCompress/app.py

```diff
@@ -0,0 +1,103 @@
+import os
+import zipfile
+import json
+import posixpath
+from pprint import pprint
+import boto3
+import shutil
+
+
+def lambda_handler(event, context):
+    # Define the constants
+    TABLE_NAME = os.environ["JOB_TABLE"]
+    BUCKET_NAME = os.environ["SRC_BUCKET"]
+    PRIMARY_KEY = "id"
+    LOOKUP_BATCH_ID = event["queryStringParameters"]["id"]
+
+    # Create table and bucket reference
+    table = boto3.resource("dynamodb").Table(TABLE_NAME)
+    bucket = boto3.resource("s3").Bucket(BUCKET_NAME)
+
+    # Get the data from DynamoDB
+    response = table.get_item(Key={PRIMARY_KEY: LOOKUP_BATCH_ID})
+
+    # Download files using reference data from DynamoDB
+    files_json = response["Item"]["files"]
+
+    # Create zip file
+    zip_file_path = os.path.join("/tmp/", "temp_zip.zip")
+    temp_zip = zipfile.ZipFile(zip_file_path, "w")
+
+    # Go through each file
+    for file in files_json:
+
+        # Get file path for s3 bucket and create file path to store in tmp
+        file_ref_s3 = file["fileRef"]
+        file_ref_tmp = "/tmp/" + file["fileRef"].split("/")[-1]
+
+        # Download the uploaded file
+        bucket.download_file(file_ref_s3, file_ref_tmp)
+
+        # Add it to the zip file in the uploaded section
+        temp_zip.write(
+            file_ref_tmp,
+            "uploaded/" + file["fileRef"].split("/")[-1],
+            compress_type=zipfile.ZIP_DEFLATED,
+        )
+
+        # Delete the upload file
+        os.remove(file_ref_tmp)
+
+        # Download the result file is it exists
+        if file["finished"]:
+
+            # Get file path references
+            file_ref_s3 = file["resultFileReference"]
+            file_ref_tmp = "/tmp/" + file["resultFileReference"].split("/")[-1]
+
+            # Download the results file
+            bucket.download_file(file_ref_s3, file_ref_tmp)
+
+            # Add it to the zip file in the results section
+            temp_zip.write(
+                file_ref_tmp,
+                "results/" + file["resultFileReference"].split("/")[-1],
+                compress_type=zipfile.ZIP_DEFLATED,
+            )
+
+            # Delete the results file
+            os.remove(file_ref_tmp)
+
+    # Close zip file
+    temp_zip.close()
+
+    # Add zip file to S3 bucket
+    file_ref_zip_s3 = LOOKUP_BATCH_ID + "/VisionResults.zip"
+    bucket.upload_file(zip_file_path, file_ref_zip_s3)
+
+    # Clean out tmp folder
+    for root, dirs, files in os.walk("/tmp"):
+        for f in files:
+            os.unlink(os.path.join(root, f))
+        for d in dirs:
+            shutil.rmtree(os.path.join(root, d))
+
```

Comment:
You should be writing that the zip is finished back to the database (this doesn't exist yet) and deleting all the files in the s3 bucket related to the report (i.e. only the zip should remain).

Reply:
This database is separate from the s3 bucket? Also, if it doesn't exist yet, should I just write to the s3 bucket (what it does now) as a placeholder?

Reply:
Oh, you mean writing back to the DynamoDB table! I'm guessing there'll be some sort of flag for each batch_id that I'll have to change?

Reply:
Additionally, when I'm writing back to the database, what happens with the file paths in the database since they do not exist anymore? Should I also get rid of those? Should I get rid of the files property altogether?

Reply:
I don't think we've decided yet, but what you have now is fine/can be updated later.